core/engine/src/vm/opcode/rest_parameter/mod.rs

```diff
@@ -20,16 +20,30 @@ impl Operation for RestParameterInit {
         let frame = context.vm.frame();
         let argument_count = frame.argument_count;
         let param_count = frame.code_block().parameter_length;
+        let register_count = frame.code_block().register_count;
 
-        let array = if argument_count >= param_count {
+        if argument_count >= param_count {
             let rest_count = argument_count - param_count + 1;
-            let args = context.vm.pop_n_values(rest_count as usize);
-            Array::create_array_from_list(args, context)
+
+            let len = context.vm.stack.len() as u32;
+            let start = (len - rest_count - register_count) as usize;
+            let end = (len - register_count) as usize;
+
+            let args = &context.vm.stack[start..end];
+
+            let array = Array::create_array_from_list(args.iter().cloned(), context);
+            context.vm.stack.splice(start..end, [array.clone().into()]);
```

Comment:
I think this should be `drain`, correct?

```suggestion
            context.vm.stack.drain(start..end);
```